polynote-frontend/polynote/data/value_repr.ts

```diff
@@ -160,9 +160,8 @@ export class DataStream extends EventTarget {
     private nextPromise?: {resolve: <T>(value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void}; // holds a Promise's `resolve` and `reject` inputs.
     private setupPromise?: Promise<Message | void>;
     private repr: StreamingDataRepr;
-    constructor(readonly path: string, private readonly originalRepr: StreamingDataRepr, mods?: TableOp[]) {
+    constructor(private readonly socket: SocketSession, private readonly originalRepr: StreamingDataRepr, mods?: TableOp[]) {
```

Comment:
In several places we now have to pass around the notebook's socket (or sometimes a reference to the notebook) instead of the path (because there's not a single SocketSession anymore)

Reply:
Hopefully my changes to make SocketSession a singleton a few months ago didn't make this change harder than it needed to be 😟

Reply:
On the contrary, made it super easy to find all the places using the socket session!